int8 agent_id
int8 vx
int8 vy